wallet-new/integration/TransactionSpecs.hs

```diff
@@ -18,19 +19,29 @@ import           Test.QuickCheck (arbitrary, generate)
 import           Util
 
 
-transactionSpecs :: WalletRef -> WalletClient IO -> Spec
-transactionSpecs wRef wc = do
-    describe "Transactions" $ do
-        it "Posted transactions appear in the index" $ do
-            genesis <- makeGenesisWallet wc
-            (fromAcct, _) <- getFirstAccountAndAddress wc genesis
+transactionSpecs :: HasCallStack => WalletClient IO -> Spec
+transactionSpecs wc = do
+    it "Estimate fees of a well-formed transaction" $ estimateTransactionFees wc
+    before (createRandomSampleWallet wc) $
```

Comment:
The use of `createRandomSampleWallet` seems to be replacing the old `sampleWallet` function, which would reuse an existing wallet. This is a breaking change in behavior. I'm curious why you decided to change this behavior?

Reply:
Ack. this is a breaking change. Yes, it makes sense to test the reuse of an existing.

The test lay-out and the organisation of the tests suggest that the Spec consists of several independent tests (that could for example run in parallel).
Coupling the tests with an underlying MVar looks not very intuitive.
The original design was really strange, I would like to know what the motivation for the original design is ?

Why reuse the same wallet for several test ?
If the idea is to test wallet reuse than it is much better to write a specific test.